### PR TITLE
Feature/core 9155 fix latest date year

### PR DESCRIPTION
**Pull request by dletran**

### CODE REVIEW

**Review by fofi — APPROVED** [review]
Can this happen with other granularity?

**Comment by dletran** [comment]
> Can this happen with other granularity?

@fofi I'll check it for month too, which probably it's the same, and I'll apply the same solution in that case

**Review by Navas23 — APPROVED** [review]

File: src/elements/GeoCalendar/GeoCalendar.vue [hunk 1]
```diff
@@ -435,12 +438,12 @@ export default {
...
         : {
-          start: unverifiedRange.end,
-          end: unverifiedRange.start
+          start: startOfQuarter(unverifiedRange.end),
+          end: endOfQuarter(unverifiedRange.start)
         }
 
-      this.fromRawDate = validatedRange.start && startOfQuarter(validatedRange.start)
-      this.toRawDate = validatedRange.end && endOfQuarter(validatedRange.end)
+      this.fromRawDate = validatedRange.start
```
Comment (fofi) [inline src/elements/GeoCalendar/GeoCalendar.vue:445]: Why changing this code?

File: src/elements/GeoCalendar/GeoCalendar.vue [hunk 2]
```diff
@@ -382,7 +384,7 @@ export default {
           this.isFromInputFieldExplicitlyFocused ||
           distanceToFromDate < distanceToToDate
 
-      const unverifiedRange = this.getUnverifiedRange(firstDayOfMonth, lastDayOfMonth, isSettingFromDate)
+      const unverifiedRange = this.getUnverifiedRange(firstDayOfMonth, possibleLastDayOfSelectedMonth, isSettingFromDate)
```
Comment (fofi) [inline src/elements/GeoCalendar/GeoCalendar.vue:387]: Should we do the same with the initial day?

**Review by fofi — CHANGES_REQUESTED** [review]
I think we should add some tests for this cases

File: src/elements/GeoCalendar/GeoCalendar.vue (hunk 2 above)
Comment (dletran) [reply src/elements/GeoCalendar/GeoCalendar.vue:387]: if the initial day is before the latest date we won't allow to select it in the picker itself

File: src/elements/GeoCalendar/GeoCalendar.vue (hunk 1 above)
Comment (dletran) [reply src/elements/GeoCalendar/GeoCalendar.vue:445]: If we apply the endOfQuarter after the validatedRange we'd use the end of the quarter even if it's after the latest date

File: src/elements/GeoCalendar/GeoCalendar.vue (hunk 2 above)
Comment (fofi) [reply src/elements/GeoCalendar/GeoCalendar.vue:387]: I mean if the firstDayOfMonth is smaller than earliestDate ?

Comment (dletran) [reply src/elements/GeoCalendar/GeoCalendar.vue:387]: ah good point! I'll have a look

**Review by fofi — COMMENTED** [review]
What about selectWeek ?

**Comment by dletran** [comment]
> What about selectWeek ?

@fofi we don't allow selecting a "non-complete" week on the picker, although we could add the verification just in case we change that in the future so everything works the same way, wdyt?

**Review by fofi — APPROVED** [review]

**Comment by fofi** [comment]
> > What about selectWeek ?
> 
> @fofi we don't allow selecting a "non-complete" week on the picker, although we could add the verification just in case we change that in the future so everything works the same way, wdyt?

Whatever you think is best, on my side is enough

**Review by fofi — APPROVED** [review]

**Pull request merged**